deepchecks/utils/abstracts/confusion_matrix_abstract.py

```diff
@@ -51,38 +57,69 @@ def create_confusion_matrix_figure(confusion_matrix_data: np.ndarray, classes_na
         2D array containing the confusion matrix.
     classes_names: List[str]
         the names of the classes to display as the axis.
-    normalize_display: bool
-        if True will also show normalized values by the true values.
+    max_num_labels_to_show : int
+        The threshold to display the maximum number of labels on the label prediction distribution pie
+        charts and display rest of the labels under "Others" category.
 
     Returns
     -------
     plotly Figure object
         confusion matrix figure
 
     """
-    if normalize_display:
-        confusion_matrix_norm = confusion_matrix_data.astype('float') / \
-                                (confusion_matrix_data.sum(axis=1)[:, np.newaxis] + np.finfo(float).eps) * 100
-        z = np.vectorize(format_number_if_not_nan)(confusion_matrix_norm)
-        text_template = '%{z}%<br>(%{text})'
-        color_bar_title = '% out of<br>True Values'
-        plot_title = 'Percent Out of True Values (Count)'
-    else:
-        z = confusion_matrix_data
-        color_bar_title = None
-        text_template = '%{text}'
-        plot_title = 'Value Count'
-
-    fig = go.Figure(data=go.Heatmap(
-        x=classes_names, y=classes_names, z=z,
-        text=confusion_matrix_data, texttemplate=text_template))
-    fig.data[0].colorbar.title = color_bar_title
-    fig.update_layout(title=plot_title)
-    fig.update_layout(height=600)
-    fig.update_xaxes(title='Predicted Value', type='category', scaleanchor='y', constrain='domain')
-    fig.update_yaxes(title='True Value', type='category', constrain='domain', autorange='reversed')
-
-    return fig
+    display = []
+    confusion_matrix_norm = confusion_matrix_data.astype('float') / \
+        (confusion_matrix_data.sum(axis=1)[:, np.newaxis] + np.finfo(float).eps) * 100
+
+    accuracy_array = np.diag(confusion_matrix_norm).round(decimals=2)
+    sorted_map = dict(sorted(dict(enumerate(accuracy_array)).items(), key=lambda item: item[1]))
+
+    worst_class_indices = [index for index, acc in sorted_map.items() if acc < MIN_ACCURACY_FOR_GOOD_CLASSES]
+
+    if len(accuracy_array) > MAX_WORST_CLASSES_TO_DISPLAY:
+        worst_class_indices = worst_class_indices[:MAX_WORST_CLASSES_TO_DISPLAY]
+
+    display_msg = textwrap.dedent(
+        f'The overall accuracy of your model is: {round(np.sum(accuracy_array)/len(accuracy_array), 2)}%.<br>'
+        f'Best accuracy achieved on samples with <b>{classes_names[np.argmax(accuracy_array)]}</b> '
+        f'label ({np.max(accuracy_array)}%).<br>Worst accuracy achieved on samples with '
+        f'<b>{classes_names[np.argmin(accuracy_array)]}</b> label ({np.min(accuracy_array)}%).<br>'
+        'Below are pie charts showing the prediction distribution for samples grouped based on their label.'
+    )
+    display.append(display_msg)
+    curr_col = 1
+    if len(worst_class_indices) > 0:
```

Comment:
would revert the if - check if len(worst_class_indices) == 0 then only return the text + would also return the first line of the text ( without the Worst accuracy achieved ...)

Reply:
Done